File: web/src/api/k8sDeployments.js
import service from '@/utils/request'

// @Tags K8sDeployment
// @Summary 创建K8sDeployment
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.K8sDeployment true "创建K8sDeployment"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /k8sDeployments/createK8sDeployment [post]
export const createK8sDeployment = (data) => {
     return service({
         url: "/k8sDeployments/createK8sDeployment",
         method: 'post',
         data
     })
 }


// @Tags K8sDeployment
// @Summary 删除K8sDeployment
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.K8sDeployment true "删除K8sDeployment"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /k8sDeployments/deleteK8sDeployment [delete]
 export const deleteK8sDeployment = (data) => {
     return service({
         url: "/k8sDeployments/deleteK8sDeployment",
         method: 'delete',
         data
     })
 }

// @Tags K8sDeployment
// @Summary 删除K8sDeployment
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除K8sDeployment"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /k8sDeployments/deleteK8sDeployment [delete]
 export const deleteK8sDeploymentByIds = (data) => {
     return service({
         url: "/k8sDeployments/deleteK8sDeploymentByIds",
         method: 'delete',
         data
     })
 }

// @Tags K8sDeployment
// @Summary 更新K8sDeployment
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.K8sDeployment true "更新K8sDeployment"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /k8sDeployments/updateK8sDeployment [put]
 export const updateK8sDeployment = (data) => {
     return service({
         url: "/k8sDeployments/updateK8sDeployment",
         method: 'put',
         data
     })
 }


// @Tags K8sDeployment
// @Summary 用id查询K8sDeployment
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.K8sDeployment true "用id查询K8sDeployment"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /k8sDeployments/findK8sDeployment [get]
 export const findK8sDeployment = (params) => {
     return service({
         url: "/k8sDeployments/findK8sDeployment",
         method: 'get',
         params
     })
 }


// @Tags K8sDeployment
// @Summary 分页获取K8sDeployment列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取K8sDeployment列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /k8sDeployments/getK8sDeploymentList [get]
 export const getK8sDeploymentList = (params) => {
     return service({
         url: "/k8sDeployments/getK8sDeploymentList",
         method: 'get',
         params
     })
 }

File: server/router/k8s_deployments.go
package router

import (
	v1 "gin-vue-devops/api/v1"
	"gin-vue-devops/middleware"

	"github.com/gin-gonic/gin"
)

func InitK8sDeploymentRouter(Router *gin.RouterGroup) {
	K8sDeploymentRouter := Router.Group("k8sDeployments").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler())
	{
		K8sDeploymentRouter.POST("createK8sDeployment", v1.CreateK8sDeployment)             // 新建K8sDeployment
		K8sDeploymentRouter.DELETE("deleteK8sDeployment", v1.DeleteK8sDeployment)           // 删除K8sDeployment
		K8sDeploymentRouter.DELETE("deleteK8sDeploymentByIds", v1.DeleteK8sDeploymentByIds) // 批量删除K8sDeployment
		K8sDeploymentRouter.PUT("updateK8sDeployment", v1.UpdateK8sDeployment)              // 更新K8sDeployment
		K8sDeploymentRouter.GET("findK8sDeployment", v1.FindK8sDeployment)                  // 根据ID获取K8sDeployment
		K8sDeploymentRouter.GET("getK8sDeploymentList", v1.GetK8sDeploymentList)            // 获取K8sDeployment列表
	}
}
